junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/MethodTestDescriptor.java

```diff
@@ -108,27 +109,31 @@ public SkipResult shouldBeSkipped(JupiterEngineExecutionContext context) throws
 	@Override
 	public JupiterEngineExecutionContext execute(JupiterEngineExecutionContext context,
 			DynamicTestExecutor dynamicTestExecutor) throws Exception {
+		invokeBetweenBeforeAndAfter(context, c -> invokeTestMethod(c, dynamicTestExecutor));
+		return context;
+	}
+
+	protected void invokeBetweenBeforeAndAfter(JupiterEngineExecutionContext context,
```

Comment:
I don't like this method name: it does not align nicely with the existing methods in that class.

Something like `invokeTest()` or `executeTest()` would make more sense here.